### ISSUE TITLE
[P0] Create CAIKit-nlp community images/CI for Open Data Hub

### ISSUE THREAD

**Issue by heyselbi** [opendatahub-io/caikit-nlp#2]

**Comment by Jooho** [comment]
@vaibhavjainwiz @xianli123 I think we don't need image/ci for caikit-nlp because we will use caikit-nlp from pypi library. Am I right?

**Comment by xianli123** [comment]
Hi, @Jooho I am sorry, but I don't have the answer. I am a UX designer and not so familiar with the question. We may need to wait for @vaibhavjainwiz 's reply.

**Comment by vaibhavjainwiz** [comment]
@Jooho I have same understanding we don't need image/ci for caikit-nlp. @danielezonca @Xaenalt Could you please confirm.

**Comment by danielezonca** [comment]
We need to decide how we prefer to proceed here: caikit-nlp is a python project so we can publish it to pypi but at the same time we need to build the actual runtime image with the entry point.
At the moment @Xaenalt created a custom [`Dockerfile`](https://github.com/Xaenalt/caikit-serving/blob/main/Dockerfile) that has a simple [`start-serving.sh` entry point](https://github.com/Xaenalt/caikit-serving/blob/main/start-serving.sh#L15)
We can keep the image creation part in another repo but I think it is easier to add the dockerfile + scripts (if necessary) directly in the caikit-nlp (Note: there is this [entry-point script](https://github.com/opendatahub-io/caikit-nlp/blob/main/runtime_template/run_with_gateway.sh) already in the repo that can be used)

Long story short: I expect we will publish caikit-nlp to pypi but we will also create the image to be used for the serving. I think we can focus directly on the image part

@Xaenalt Have you already discussed with caikit team if/when they plan to publish caikit-nlp to pypi?

**Comment by Xaenalt** [comment]
We do need a CI image for the caikit runtime, which includes caikit-nlp

Separately, caikit-nlp will eventually be pushed to pypi (there are some ongoing efforts around this work, check the IBM slack)

**Comment by Xaenalt** [comment]
As for the entrypoint script, I only took the smaller part for two reasons: one is that it was a test image, and two is that they are actively refactoring the handling of connections, https://github.com/caikit/caikit/pull/268 is the issue for that. Based on that work, the start script will need some reworking, for now it's just an example of how to launch it in a minimal configuration
